src/main/webapp/javascript/user-page.js

```diff
@@ -30,10 +35,44 @@ async function loadUserRecipes() {
           elementsToAddToImageDiv.forEach(elem => imageDivs[i].appendChild(elem));
         }
       }
+      addDeleteFunctionality(recipes);
+      // TODO: add edit functionality
     }
   }
 }
 
+// adds delete functionality to the delete button in the recipe cards
+function addDeleteFunctionality(recipes){
+  const deleteButtons = document.getElementsByClassName('fa-trash-alt');
+  const recipeCards = document.getElementsByClassName('recipe-card');
+  // there are as many delete buttons as there are recipe cards
+  for (let i = 0; i < deleteButtons.length; i++) {
+    let recipe = recipes[i];
+    let recipeCard = recipeCards[i];
+    deleteButtons[i].addEventListener('click', () => {
+      const deleteConfirmed = confirm("Are you sure you want to delete the " 
+        + recipe.name + " recipe?\nThis action cannot be undone!");
+      if (deleteConfirmed) {
+        deleteRecipe(recipe);
+        // Remove the recipe from the DOM.
+        recipeCard.remove();
+      }
+      if (recipeCards.length == 0) {
+        noRecipes("user-recipes", NO_USER_RECIPES);
+      }
+    });
+  }
+}
+
+// takes in div id and message
+// makes the recipes container bigger and gives it the message
+// to use when there are no recipes in planner/cookbook/user-recipes
+noRecipes = (divID, message) => {
```

Comment:
Can you give this a more descriptive name? `noRecipes` indicates the situation but not what the function is actually doing, something like `setupDivWithNoRecipes` or `setNoRecipesMessage` I think would be more clear

Reply:
okay sounds good!